### ISSUE TITLE
TCH002 false positive for certain target_version or no `__future__.annotations` import

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#3299]
I fixed my code according to `TCH002` and got a `NameError` exception when executing it and the code doesn't have the `from __future__ import annotations` import.

```python
#!/usr/bin/python3

#from __future__ import annotations

from typing import Iterator, TYPE_CHECKING

from apt_pkg import Version

if TYPE_CHECKING:
    from apt_pkg import Version2

class Foo:

    def bar(self) -> Iterator[Version]:
        pkg = self.apt['foo']
        cand = pkg.candidate
        yield cand

    def baz(self) -> Iterator[Version2]:
        pkg = self.apt['foo']
        cand = pkg.candidate
        yield cand

Foo().bar()
Foo().baz()

```
→
```console
$ python3.9 foo.py
Traceback (most recent call last):
  File "foo.py", line 12, in <module>
    class Foo:
  File "foo.py", line 19, in Foo
    def baz(self) -> Iterator[Version2]:
NameError: name 'Version2' is not defined
```

**Comment by charliermarsh** [comment]
👍 Those should be treated as runtime annotations, since they're defined within a class scope. Thank you!
